distros/M/Math-GSL/pm/Math/GSL/Siman.pm.2.3
# This file was automatically generated by SWIG (http://www.swig.org).
# Version 3.0.4
#
# Do not make changes to this file unless you know what you are doing--modify
# the SWIG interface file instead.

package Math::GSL::Siman;
use base qw(Exporter);
use base qw(DynaLoader);
package Math::GSL::Simanc;
bootstrap Math::GSL::Siman;
package Math::GSL::Siman;
@EXPORT = qw();

# ---------- BASE METHODS -------------

package Math::GSL::Siman;

sub TIEHASH {
    my ($classname,$obj) = @_;
    return bless $obj, $classname;
}

sub CLEAR { }

sub FIRSTKEY { }

sub NEXTKEY { }

sub FETCH {
    my ($self,$field) = @_;
    my $member_func = "swig_${field}_get";
    $self->$member_func();
}

sub STORE {
    my ($self,$field,$newval) = @_;
    my $member_func = "swig_${field}_set";
    $self->$member_func($newval);
}

sub this {
    my $ptr = shift;
    return tied(%$ptr);
}


# ------- FUNCTION WRAPPERS --------

package Math::GSL::Siman;

*gsl_error = *Math::GSL::Simanc::gsl_error;
*gsl_stream_printf = *Math::GSL::Simanc::gsl_stream_printf;
*gsl_strerror = *Math::GSL::Simanc::gsl_strerror;
*gsl_set_error_handler = *Math::GSL::Simanc::gsl_set_error_handler;
*gsl_set_error_handler_off = *Math::GSL::Simanc::gsl_set_error_handler_off;
*gsl_set_stream_handler = *Math::GSL::Simanc::gsl_set_stream_handler;
*gsl_set_stream = *Math::GSL::Simanc::gsl_set_stream;
*gsl_siman_solve = *Math::GSL::Simanc::gsl_siman_solve;
*gsl_siman_solve_many = *Math::GSL::Simanc::gsl_siman_solve_many;

############# Class : Math::GSL::Siman::gsl_siman_params_t ##############

package Math::GSL::Siman::gsl_siman_params_t;
use vars qw(@ISA %OWNER %ITERATORS %BLESSEDMEMBERS);
@ISA = qw( Math::GSL::Siman );
%OWNER = ();
%ITERATORS = ();
*swig_n_tries_get = *Math::GSL::Simanc::gsl_siman_params_t_n_tries_get;
*swig_n_tries_set = *Math::GSL::Simanc::gsl_siman_params_t_n_tries_set;
*swig_iters_fixed_T_get = *Math::GSL::Simanc::gsl_siman_params_t_iters_fixed_T_get;
*swig_iters_fixed_T_set = *Math::GSL::Simanc::gsl_siman_params_t_iters_fixed_T_set;
*swig_step_size_get = *Math::GSL::Simanc::gsl_siman_params_t_step_size_get;
*swig_step_size_set = *Math::GSL::Simanc::gsl_siman_params_t_step_size_set;
*swig_k_get = *Math::GSL::Simanc::gsl_siman_params_t_k_get;
*swig_k_set = *Math::GSL::Simanc::gsl_siman_params_t_k_set;
*swig_t_initial_get = *Math::GSL::Simanc::gsl_siman_params_t_t_initial_get;
*swig_t_initial_set = *Math::GSL::Simanc::gsl_siman_params_t_t_initial_set;
*swig_mu_t_get = *Math::GSL::Simanc::gsl_siman_params_t_mu_t_get;
*swig_mu_t_set = *Math::GSL::Simanc::gsl_siman_params_t_mu_t_set;
*swig_t_min_get = *Math::GSL::Simanc::gsl_siman_params_t_t_min_get;
*swig_t_min_set = *Math::GSL::Simanc::gsl_siman_params_t_t_min_set;
sub new {
    my $pkg = shift;
    my $self = Math::GSL::Simanc::new_gsl_siman_params_t(@_);
    bless $self, $pkg if defined($self);
}

sub DESTROY {
    return unless $_[0]->isa('HASH');
    my $self = tied(%{$_[0]});
    return unless defined $self;
    delete $ITERATORS{$self};
    if (exists $OWNER{$self}) {
        Math::GSL::Simanc::delete_gsl_siman_params_t($self);
        delete $OWNER{$self};
    }
}

sub DISOWN {
    my $self = shift;
    my $ptr = tied(%$self);
    delete $OWNER{$ptr};
}

sub ACQUIRE {
    my $self = shift;
    my $ptr = tied(%$self);
    $OWNER{$ptr} = 1;
}


# ------- VARIABLE STUBS --------

package Math::GSL::Siman;

*GSL_MAJOR_VERSION = *Math::GSL::Simanc::GSL_MAJOR_VERSION;
*GSL_MINOR_VERSION = *Math::GSL::Simanc::GSL_MINOR_VERSION;
*GSL_POSZERO = *Math::GSL::Simanc::GSL_POSZERO;
*GSL_NEGZERO = *Math::GSL::Simanc::GSL_NEGZERO;
*GSL_SUCCESS = *Math::GSL::Simanc::GSL_SUCCESS;
*GSL_FAILURE = *Math::GSL::Simanc::GSL_FAILURE;
*GSL_CONTINUE = *Math::GSL::Simanc::GSL_CONTINUE;
*GSL_EDOM = *Math::GSL::Simanc::GSL_EDOM;
*GSL_ERANGE = *Math::GSL::Simanc::GSL_ERANGE;
*GSL_EFAULT = *Math::GSL::Simanc::GSL_EFAULT;
*GSL_EINVAL = *Math::GSL::Simanc::GSL_EINVAL;
*GSL_EFAILED = *Math::GSL::Simanc::GSL_EFAILED;
*GSL_EFACTOR = *Math::GSL::Simanc::GSL_EFACTOR;
*GSL_ESANITY = *Math::GSL::Simanc::GSL_ESANITY;
*GSL_ENOMEM = *Math::GSL::Simanc::GSL_ENOMEM;
*GSL_EBADFUNC = *Math::GSL::Simanc::GSL_EBADFUNC;
*GSL_ERUNAWAY = *Math::GSL::Simanc::GSL_ERUNAWAY;
*GSL_EMAXITER = *Math::GSL::Simanc::GSL_EMAXITER;
*GSL_EZERODIV = *Math::GSL::Simanc::GSL_EZERODIV;
*GSL_EBADTOL = *Math::GSL::Simanc::GSL_EBADTOL;
*GSL_ETOL = *Math::GSL::Simanc::GSL_ETOL;
*GSL_EUNDRFLW = *Math::GSL::Simanc::GSL_EUNDRFLW;
*GSL_EOVRFLW = *Math::GSL::Simanc::GSL_EOVRFLW;
*GSL_ELOSS = *Math::GSL::Simanc::GSL_ELOSS;
*GSL_EROUND = *Math::GSL::Simanc::GSL_EROUND;
*GSL_EBADLEN = *Math::GSL::Simanc::GSL_EBADLEN;
*GSL_ENOTSQR = *Math::GSL::Simanc::GSL_ENOTSQR;
*GSL_ESING = *Math::GSL::Simanc::GSL_ESING;
*GSL_EDIVERGE = *Math::GSL::Simanc::GSL_EDIVERGE;
*GSL_EUNSUP = *Math::GSL::Simanc::GSL_EUNSUP;
*GSL_EUNIMPL = *Math::GSL::Simanc::GSL_EUNIMPL;
*GSL_ECACHE = *Math::GSL::Simanc::GSL_ECACHE;
*GSL_ETABLE = *Math::GSL::Simanc::GSL_ETABLE;
*GSL_ENOPROG = *Math::GSL::Simanc::GSL_ENOPROG;
*GSL_ENOPROGJ = *Math::GSL::Simanc::GSL_ENOPROGJ;
*GSL_ETOLF = *Math::GSL::Simanc::GSL_ETOLF;
*GSL_ETOLX = *Math::GSL::Simanc::GSL_ETOLX;
*GSL_ETOLG = *Math::GSL::Simanc::GSL_ETOLG;
*GSL_EOF = *Math::GSL::Simanc::GSL_EOF;

@EXPORT_OK = qw/
               gsl_siman_solve
               gsl_siman_solve_many
             /;
%EXPORT_TAGS = ( all => [ @EXPORT_OK ] );

__END__

=encoding utf8

=head1 NAME

Math::GSL::Siman - Simulated Annealing

=head1 SYNOPSIS

This module is not yet implemented. Patches Welcome!

use Math::GSL::Siman qw /:all/;

=head1 DESCRIPTION

Here is a list of all the functions in this module :

=over

=item * C<gsl_siman_solve >

=item * C<gsl_siman_solve_many >

=back


For more informations on the functions, we refer you to the GSL offcial
documentation: L<http://www.gnu.org/software/gsl/manual/html_node/>




=head1 AUTHORS

Jonathan "Duke" Leto <jonathan@leto.net> and Thierry Moisan <thierry.moisan@gmail.com>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2008-2011 Jonathan "Duke" Leto and Thierry Moisan

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.

=cut



1;


distros/M/Math-GSL/lib/Math/GSL/Siman.pm
# This file was automatically generated by SWIG (http://www.swig.org).
# Version 2.0.12
#
# Do not make changes to this file unless you know what you are doing--modify
# the SWIG interface file instead.

package Math::GSL::Siman;
use base qw(Exporter);
use base qw(DynaLoader);
package Math::GSL::Simanc;
bootstrap Math::GSL::Siman;
package Math::GSL::Siman;
@EXPORT = qw();

# ---------- BASE METHODS -------------

package Math::GSL::Siman;

sub TIEHASH {
    my ($classname,$obj) = @_;
    return bless $obj, $classname;
}

sub CLEAR { }

sub FIRSTKEY { }

sub NEXTKEY { }

sub FETCH {
    my ($self,$field) = @_;
    my $member_func = "swig_${field}_get";
    $self->$member_func();
}

sub STORE {
    my ($self,$field,$newval) = @_;
    my $member_func = "swig_${field}_set";
    $self->$member_func($newval);
}

sub this {
    my $ptr = shift;
    return tied(%$ptr);
}


# ------- FUNCTION WRAPPERS --------

package Math::GSL::Siman;

*gsl_error = *Math::GSL::Simanc::gsl_error;
*gsl_stream_printf = *Math::GSL::Simanc::gsl_stream_printf;
*gsl_strerror = *Math::GSL::Simanc::gsl_strerror;
*gsl_set_error_handler = *Math::GSL::Simanc::gsl_set_error_handler;
*gsl_set_error_handler_off = *Math::GSL::Simanc::gsl_set_error_handler_off;
*gsl_set_stream_handler = *Math::GSL::Simanc::gsl_set_stream_handler;
*gsl_set_stream = *Math::GSL::Simanc::gsl_set_stream;
*gsl_siman_solve = *Math::GSL::Simanc::gsl_siman_solve;
*gsl_siman_solve_many = *Math::GSL::Simanc::gsl_siman_solve_many;

############# Class : Math::GSL::Siman::gsl_siman_params_t ##############

package Math::GSL::Siman::gsl_siman_params_t;
use vars qw(@ISA %OWNER %ITERATORS %BLESSEDMEMBERS);
@ISA = qw( Math::GSL::Siman );
%OWNER = ();
%ITERATORS = ();
*swig_n_tries_get = *Math::GSL::Simanc::gsl_siman_params_t_n_tries_get;
*swig_n_tries_set = *Math::GSL::Simanc::gsl_siman_params_t_n_tries_set;
*swig_iters_fixed_T_get = *Math::GSL::Simanc::gsl_siman_params_t_iters_fixed_T_get;
*swig_iters_fixed_T_set = *Math::GSL::Simanc::gsl_siman_params_t_iters_fixed_T_set;
*swig_step_size_get = *Math::GSL::Simanc::gsl_siman_params_t_step_size_get;
*swig_step_size_set = *Math::GSL::Simanc::gsl_siman_params_t_step_size_set;
*swig_k_get = *Math::GSL::Simanc::gsl_siman_params_t_k_get;
*swig_k_set = *Math::GSL::Simanc::gsl_siman_params_t_k_set;
*swig_t_initial_get = *Math::GSL::Simanc::gsl_siman_params_t_t_initial_get;
*swig_t_initial_set = *Math::GSL::Simanc::gsl_siman_params_t_t_initial_set;
*swig_mu_t_get = *Math::GSL::Simanc::gsl_siman_params_t_mu_t_get;
*swig_mu_t_set = *Math::GSL::Simanc::gsl_siman_params_t_mu_t_set;
*swig_t_min_get = *Math::GSL::Simanc::gsl_siman_params_t_t_min_get;
*swig_t_min_set = *Math::GSL::Simanc::gsl_siman_params_t_t_min_set;
sub new {
    my $pkg = shift;
    my $self = Math::GSL::Simanc::new_gsl_siman_params_t(@_);
    bless $self, $pkg if defined($self);
}

sub DESTROY {
    return unless $_[0]->isa('HASH');
    my $self = tied(%{$_[0]});
    return unless defined $self;
    delete $ITERATORS{$self};
    if (exists $OWNER{$self}) {
        Math::GSL::Simanc::delete_gsl_siman_params_t($self);
        delete $OWNER{$self};
    }
}

sub DISOWN {
    my $self = shift;
    my $ptr = tied(%$self);
    delete $OWNER{$ptr};
}

sub ACQUIRE {
    my $self = shift;
    my $ptr = tied(%$self);
    $OWNER{$ptr} = 1;
}


# ------- VARIABLE STUBS --------

package Math::GSL::Siman;

*GSL_MAJOR_VERSION = *Math::GSL::Simanc::GSL_MAJOR_VERSION;
*GSL_MINOR_VERSION = *Math::GSL::Simanc::GSL_MINOR_VERSION;
*GSL_POSZERO = *Math::GSL::Simanc::GSL_POSZERO;
*GSL_NEGZERO = *Math::GSL::Simanc::GSL_NEGZERO;
*GSL_SUCCESS = *Math::GSL::Simanc::GSL_SUCCESS;
*GSL_FAILURE = *Math::GSL::Simanc::GSL_FAILURE;
*GSL_CONTINUE = *Math::GSL::Simanc::GSL_CONTINUE;
*GSL_EDOM = *Math::GSL::Simanc::GSL_EDOM;
*GSL_ERANGE = *Math::GSL::Simanc::GSL_ERANGE;
*GSL_EFAULT = *Math::GSL::Simanc::GSL_EFAULT;
*GSL_EINVAL = *Math::GSL::Simanc::GSL_EINVAL;
*GSL_EFAILED = *Math::GSL::Simanc::GSL_EFAILED;
*GSL_EFACTOR = *Math::GSL::Simanc::GSL_EFACTOR;
*GSL_ESANITY = *Math::GSL::Simanc::GSL_ESANITY;
*GSL_ENOMEM = *Math::GSL::Simanc::GSL_ENOMEM;
*GSL_EBADFUNC = *Math::GSL::Simanc::GSL_EBADFUNC;
*GSL_ERUNAWAY = *Math::GSL::Simanc::GSL_ERUNAWAY;
*GSL_EMAXITER = *Math::GSL::Simanc::GSL_EMAXITER;
*GSL_EZERODIV = *Math::GSL::Simanc::GSL_EZERODIV;
*GSL_EBADTOL = *Math::GSL::Simanc::GSL_EBADTOL;
*GSL_ETOL = *Math::GSL::Simanc::GSL_ETOL;
*GSL_EUNDRFLW = *Math::GSL::Simanc::GSL_EUNDRFLW;
*GSL_EOVRFLW = *Math::GSL::Simanc::GSL_EOVRFLW;
*GSL_ELOSS = *Math::GSL::Simanc::GSL_ELOSS;
*GSL_EROUND = *Math::GSL::Simanc::GSL_EROUND;
*GSL_EBADLEN = *Math::GSL::Simanc::GSL_EBADLEN;
*GSL_ENOTSQR = *Math::GSL::Simanc::GSL_ENOTSQR;
*GSL_ESING = *Math::GSL::Simanc::GSL_ESING;
*GSL_EDIVERGE = *Math::GSL::Simanc::GSL_EDIVERGE;
*GSL_EUNSUP = *Math::GSL::Simanc::GSL_EUNSUP;
*GSL_EUNIMPL = *Math::GSL::Simanc::GSL_EUNIMPL;
*GSL_ECACHE = *Math::GSL::Simanc::GSL_ECACHE;
*GSL_ETABLE = *Math::GSL::Simanc::GSL_ETABLE;
*GSL_ENOPROG = *Math::GSL::Simanc::GSL_ENOPROG;
*GSL_ENOPROGJ = *Math::GSL::Simanc::GSL_ENOPROGJ;
*GSL_ETOLF = *Math::GSL::Simanc::GSL_ETOLF;
*GSL_ETOLX = *Math::GSL::Simanc::GSL_ETOLX;
*GSL_ETOLG = *Math::GSL::Simanc::GSL_ETOLG;
*GSL_EOF = *Math::GSL::Simanc::GSL_EOF;

@EXPORT_OK = qw/
               gsl_siman_solve
               gsl_siman_solve_many
             /;
%EXPORT_TAGS = ( all => [ @EXPORT_OK ] );

__END__

=encoding utf8

=head1 NAME

Math::GSL::Siman - Simulated Annealing

=head1 SYNOPSIS

This module is not yet implemented. Patches Welcome!

use Math::GSL::Siman qw /:all/;

=head1 DESCRIPTION

Here is a list of all the functions in this module :

=over

=item * C<gsl_siman_solve >

=item * C<gsl_siman_solve_many >

=back


For more informations on the functions, we refer you to the GSL offcial
documentation: L<http://www.gnu.org/software/gsl/manual/html_node/>




=head1 AUTHORS

Jonathan "Duke" Leto <jonathan@leto.net> and Thierry Moisan <thierry.moisan@gmail.com>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2008-2011 Jonathan "Duke" Leto and Thierry Moisan

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.

=cut



1;
